src/constants.rs

```diff
@@ -423,17 +423,23 @@ pub enum IndexSignatureTag {
     RPMTAG_INSTALLTIME = IndexTag::RPMTAG_INSTALLTIME as u32,
 }
 
-pub trait TypeName {
```

Comment:
`TypeName` is only used for `Tag`, and there are only 2 implementers of `Tag`, so we can ditch the blanket impl and `TypeName` trait.